pkg/logger/worker.go

```diff
@@ -0,0 +1,122 @@
+package logger
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudevents/sdk-go"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport"
+	"github.com/go-logr/logr"
+	"net/http"
+	"time"
+)
+
+const (
+	CEInferenceRequest  = "org.kubeflow.serving.inference.request"
+	CEInferenceResponse = "org.kubeflow.serving.inference.response"
+	ModelIdHeader       = "KF-Model-ID"
+)
+
+// NewWorker creates, and returns a new Worker object. Its only argument
+// is a channel that the worker can add itself to whenever it is done its
+// work.
+func NewWorker(id int, workerQueue chan chan LogRequest, log logr.Logger) Worker {
+	// Create, and return the worker.
+	worker := Worker{
+		Log:         log,
+		ID:          id,
+		Work:        make(chan LogRequest),
+		WorkerQueue: workerQueue,
+		QuitChan:    make(chan bool),
+		Client: http.Client{
+			Timeout: 60 * time.Second,
+		},
+		CeCtx: cloudevents.ContextWithEncoding(context.Background(), cloudevents.Binary),
+	}
+
+	return worker
+}
+
+type Worker struct {
+	Log         logr.Logger
+	ID          int
+	Work        chan LogRequest
+	WorkerQueue chan chan LogRequest
+	QuitChan    chan bool
+	Client      http.Client
+	CeCtx       context.Context
+	CeTransport transport.Transport
+}
+
+func (W *Worker) sendCloudEvent(logReq LogRequest) error {
+
+	t, err := cloudevents.NewHTTPTransport(
+		cloudevents.WithTarget(logReq.url.String()),
+		cloudevents.WithEncoding(cloudevents.HTTPBinaryV01),
+		cloudevents.WitHHeader(ModelIdHeader, logReq.modelId),
+		//cloudevents.WithContextBasedEncoding(), // toggle this or WithEncoding to see context based encoding work.
+	)
+
+	if err != nil {
+		return err
```

Comment:
I've recently adopted a new pattern when writing go for annotating errors.

return fmt.Errorf("while creating http transport: %w, err")

The idea is to use lower cased contextualized strings so that eventually you might get:

EventLogger failed "sending cloud event" "while creating http transport": target does not exist

I basically annotate all my `return err` statements now.

Reply:
%w doesn't seem to exist will use %s